### PR TITLE
Add espeakup and pyttsx3 packages

### PR DESCRIPTION
**Pull request by ethindp**
###### Description of changes

This PR adds two packages:

* espeakup: espeakup is a console-based screen reader that bridges the espeak-ng package and the speakup Linux kernel modules, which implement the actual screen reading functionality. The screen reader is minimal: it resides as a kernel module (speakup_soft, or any number of other speech synthesis modules) and, through the module you choose to load, communicates with the given speech synthesis engine. espeakup connects to speakup to provide software-based speech synthesis functionality.
* pyttsx3: see https://github.com/nateshmbhat/pyttsx3

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:25]: You need to add yourself to `maintainers/maintainer-list.nix` first.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:19]: `mesonFlags` should be a list.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:15]: `alsaLib` and `systemd` should probably got to `buildInputs` (if they are runtime, rather than build-time dependencies).

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:24]: `gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

**Comment by ethindp** [comment]
All done.

**Comment by ethindp** [comment]
cc @jtojnar

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [reply pkgs/applications/accessibility/espeakup/default.nix:15]: Looks like the dependencies are still in `nativeBuildInputs`.

**Comment by jtojnar** [comment]
Could you please make the maintainer addition a separate commit following the contribution guide (i.e. the commit message should say something like `maintainers: add ethindp`)

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:15]: `alsaLib` is an alias of `alsa-lib` so this will not build with `nixpkgs.config.allowAliases = false;`.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:21]: Missing `meta.platforms` – unlike `python3.pkgs.buildPythonPackage`, `stdenv.mkDerivation` does not set it by default.

**Comment by ethindp** [comment]
I just reworded one of my commits -- I can't really make it a "separate" commit without reverting the file. I hope that's okay.

File: pkgs/development/python-modules/pyttsx3/default.nix
Comment (jtojnar) [inline pkgs/development/python-modules/pyttsx3/default.nix:1]: Unlike top-level packages, Python packages in `pkgs/development/python-modules` should get dependencies from the `python3.pkgs` scope as arguments directly – that means do not pass `python3Packages` here.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:15]: Are you sure `cmake` is needed? Meson supports it for finding dependencies but typically, `pkg-config` will be sufficient. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/meson-cmake.md> for more info.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:9]: Please use `"v${version}"` so that it can be automatically updated.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:10]: No need to double the sha256 here.

File: pkgs/development/python-modules/pyttsx3/default.nix
Comment (jtojnar) [inline pkgs/development/python-modules/pyttsx3/default.nix:10]: You can use `inherit format;` here to avoid repetition.

File: pkgs/development/python-modules/pyttsx3/default.nix
Comment (jtojnar) [inline pkgs/development/python-modules/pyttsx3/default.nix:1]: `ffmpeg` does not seem to be used.

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:19]: Apparently, original ronn is unmaintained and this requires a [fork](https://github.com/apjanke/ronn-ng). Could you please add a comment with explanation?

```suggestion
  mesonFlags = [
    # depends on fork of ronn not available in Nixpkgs
    "-Dman=disabled"
  ];
```

File: pkgs/applications/accessibility/espeakup/default.nix
Comment (jtojnar) [inline pkgs/applications/accessibility/espeakup/default.nix:23]: Descriptions should not start with an article (or space, for that matter).

File: pkgs/development/python-modules/pyttsx3/default.nix
Comment (jtojnar) [inline pkgs/development/python-modules/pyttsx3/default.nix:15]: Please add comment why this is disabled.

File: pkgs/development/python-modules/pyttsx3/default.nix
Comment (jtojnar) [inline pkgs/development/python-modules/pyttsx3/default.nix:18]: This is probably too unweildly, I would use something like “Text to Speech library for Python”. Or “Offline Text To Speech synthesis for Python”, which they use on their GitHub. We can always put the rest into `longDescription`.

File: pkgs/development/python-modules/pyttsx3/default.nix
Comment (jtojnar) [reply pkgs/development/python-modules/pyttsx3/default.nix:14]: Actually, looks like the comment about `doCheck = false` is still missing.

**Review by jtojnar — APPROVED** [review]

**Comment by jtojnar** [comment]
Sorry, forgot about this. Since no-one else reviewed, we can merge this after rebasing to resolve the conflicts.

**Comment by ethindp** [comment]
I (think) I fixed those, unless my recent updates to my fork caused issues?

**Comment by jtojnar** [comment]
Back-merges make the history more convoluted so it is preferred to avoid them in PRs.

I have rebased this onto master and moved the entry in maintainer-list.nix to the alphabetically correct place to reduce the risk of future collisions. And looks like I can push to your branch after all, it is just that `gh` tool sets the branch for pushing incorrectly.

**Comment by ethindp** [comment]
The back-merges were because I thought this PR had already been closed. I hadn't realized it was still open.

**Comment by jtojnar** [comment]
Ronn update was merged so I enabled the man page.

**Review by jtojnar — APPROVED** [review]

**Pull request merged**
